hal/watchdog_api.h

```diff
@@ -0,0 +1,149 @@
+/** \addtogroup hal */
+/** @{*/
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_WATCHDOG_API_H
+#define MBED_WATCHDOG_API_H
+
+#if DEVICE_WATCHDOG
+
+#include <stdbool.h>
+#include <stdint.h>
+
+/**
+ * \defgroup hal_watchdog Watchdog HAL API
+ * @{
+ */
+
+/** \file watchdog_api.h
```

Comment:
Could we copy the defined/undefined APIs from the handbook to match other hal apis.

Reply:
Added in https://github.com/ARMmbed/mbed-os/pull/10857/commits/a33536e78e9efe266eaa44ce6e4ec8b416b64bcd